src/Umbraco.Infrastructure/Migrations/Upgrade/V_10_5_0/AddPrimaryKeyConstrainToContentVersionCleanupDtos.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Umbraco.Cms.Infrastructure.Persistence.Dtos;
+
+namespace Umbraco.Cms.Infrastructure.Migrations.Upgrade.V_10_5_0;
+
+public class AddPrimaryKeyConstrainToContentVersionCleanupDtos : MigrationBase
+{
+    public AddPrimaryKeyConstrainToContentVersionCleanupDtos(IMigrationContext context) : base(context)
+    {
+    }
+
+    protected override void Migrate()
+    {
+        IEnumerable<ContentVersionCleanupPolicyDto> contentVersionCleanupPolicyDtos =
+            Database
+                .Fetch<ContentVersionCleanupPolicyDto>()
+                .OrderByDescending(x => x.Updated)
+                .DistinctBy(x => x.ContentTypeId);
+
+        if (TableExists(ContentVersionCleanupPolicyDto.TableName))
```

Comment:
@Zeegaan Won't fetching data from a table that doesn't exist throw an exception already, making the `TableExists` call redundant?

Reply:
And if the primary key index already exists, this migration shouldn't have to get all data, delete and re-create the table and insert all data again...

This could be achieved by setting an explicit name on the `PrimaryKeyColumn` attribute (e.g. ``PK_umbracoContentVersionCleanupPolicy`) and checking whether this exists using `IndexExists` in this migration.

Reply:
Hmm you are probably right, we should remedy that 😅